taf/repository_tool.py

```diff
@@ -65,6 +65,18 @@ def is_delegated_role(role):
     return role not in ("root", "targets", "snapshot", "timestamp")
 
 
+def is_auth_repo(repo_path):
```

Comment:
We rely on TUF to check if the folder contains a valid repository. TUF raises an error if the metadata directory exists, but targets does not. I am not sure when exactly we have the case of metadata directory existing and targets not, that instantiation of TUF's repository class is among oldest parts of TAF code. It would make sense to make sure that we are generating both metadata and targets directories when, say, creating a new authentication repository using TAF's cli. Since we do not have developer tool tests yet, I would prefer not to touch that code for now. I think that it makes sense to move removal of targets directory to the `except` block of the method which creates it if it does not exist.